src/Utilities/Options/CategorizedAnalyzerConfigOptions.cs

```diff
@@ -113,7 +113,7 @@ public T GetOptionValue<T>(string optionName, DiagnosticDescriptor rule, TryPars
                 return defaultValue;
             }
 
-            return (T)_computedOptionValuesMap.GetOrAdd(optionName, _ => ComputeOptionValue(optionName, rule, tryParseValue, defaultValue));
+            return (T)_computedOptionValuesMap.GetOrAdd($"{rule.Id}.{optionName}", _ => ComputeOptionValue(optionName, rule, tryParseValue, defaultValue));
```

Comment:
So this would make the size of the map O((rule count) * (option count))?

Reply:
Yes, but practically not that big as number of supported and queried combinations are pretty less.